cms/core/models/snippets.py

```diff
@@ -53,3 +58,61 @@ def save(self, *args: Any, **kwargs: Any) -> None:
 
     def __str__(self) -> str:
         return str(self.name)
+
+
+class GlossaryTerm(TranslatableMixin, PreviewableMixin, RevisionMixin, index.Indexed, models.Model):
+    """A model for glossary terms."""
+
+    title = models.CharField(max_length=255)
+    definition = RichTextField(features=settings.RICH_TEXT_BASIC)
+    # to allow for a user to be set on creation and seen in the IndexView
+    updated_by = models.ForeignKey(
+        "users.User",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="glossary_terms",
+    )
+    # See https://docs.wagtail.org/en/stable/advanced_topics/reference_index.html
+    updated_by.wagtail_reference_index_ignore = True  # type: ignore[attr-defined]
+
+    revisions = GenericRelation("wagtailcore.Revision", related_query_name="glossary_term")
+
+    panels: ClassVar[list["Panel"]] = [
+        "title",
+        "definition",
+    ]
+
+    search_fields: ClassVar[list[index.BaseField]] = [
+        *index.Indexed.search_fields,
+        index.SearchField("title"),
+        index.AutocompleteField("title"),
+        index.SearchField("definition"),
+        index.AutocompleteField("definition"),
+    ]
+
+    class Meta:
+        verbose_name = _("glossary term")
+        verbose_name_plural = _("glossary terms")
+
+        constraints: ClassVar[list[models.BaseConstraint]] = [
+            models.UniqueConstraint(
+                Lower("title"),
+                name="core_glossary_term_title_unique",
+                violation_error_message=_("A glossary term with this title already exists."),
+            ),
+            models.UniqueConstraint(
+                fields=("translation_key", "locale"), name="unique_translation_key_locale_core_glossaryterm"
+            ),
+        ]
+
+    def get_preview_template(self, request: "HttpRequest", mode_name: str) -> str:
+        return "templates/components/glossary/glossary_term_preview.html"
+
+    def __str__(self) -> str:
+        return str(self.title)
+
+    def save(self, *args: Any, **kwargs: Any) -> None:
+        if self.title:
```

Comment:
would there ever be a case where there isn't a title, given that it is a required field?

Reply:
I think I just copied that from the ContactDetails snippet without thinking much. 
In https://github.com/ONSdigital/dis-wagtail/pull/110/commits/b8bb7a7c6b099cac41b09b2e205ba7b25d7c240c I removed the condition for `save` functions for both snippets (since the `name` in ContactDetails is also a mandatory field)